cluster-api-gcp/deploy/machinedeployer.go

```diff
@@ -0,0 +1,33 @@
+package deploy
+
+import (
+	"fmt"
+
+	clusterv1 "k8s.io/kube-deploy/cluster-api/api/cluster/v1alpha1"
+	"k8s.io/kube-deploy/cluster-api-gcp/cloud"
+	"k8s.io/kube-deploy/cluster-api-gcp/cloud/google"
+)
+
+// Provider-specific machine logic the deployer needs.
+type machineDeployer interface {
+	cloud.MachineActuator
+	GetIP(machine *clusterv1.Machine) (string, error)
+	GetKubeConfig(master *clusterv1.Machine) (string, error)
+
+	// Create and start the machine controller. The list of initial
+	// machines don't have to be reconciled as part of this function, but
+	// are provided in case the function wants to refer to them (and their
+	// ProviderConfigs) to know how to configure the machine controller.
+	// Not idempotent.
+	CreateMachineController(cluster *clusterv1.Cluster, initialMachines []*clusterv1.Machine) error
+	PostDelete(cluster *clusterv1.Cluster, machines []*clusterv1.Machine) error
+}
+
+func newMachineDeployer(cloud string, kubeadmToken string) (machineDeployer,error) {
```

Comment:
since this code is in the gcp directory, do we really need a switch statement?

Reply:
The original machine controller code was written to separate the common cloud-agnostic code from the cloud-specific code. The goal is that if the common code is common enough, we would promote that back into the cluster-api. 

As such, the cloud string can have different values including google, test etc. See the following for example values that might come in:
https://github.com/kubernetes/kube-deploy/blob/master/cluster-api-gcp/cloud/actuators.go#L38

Since google is not the only option, we should error out for all options we do not yet support.